### PR TITLE
Move some tests to doctest

### PR DESCRIPTION
**Pull request by eivindjahren**
As part of cleaning up the tests (from retrospective), we need better documentation of what the tests are trying to verify. This PR refactors some tests to doctest so that it is clear why these checks exist.

## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

File: src/ert/_c_wrappers/enkf/ensemble_config.py [hunk 1]
```diff
@@ -31,6 +31,35 @@ def _get_filename(file):
...
+
+    From there, _option_dict can be used to get a dictionary of the options:
+
+    >>> _option_dict(opts, 1)
+    {'INPUT_FORMAT': 'ASCII', 'RESULT_FILE': 'file.txt', 'REPORT_STEPS': '3'}
+
+    Errors are reported to the log, and erroring fields ignored:
+
+    >>> _option_dict(opts, 1)
+    {'INPUT_FORMAT': 'ASCII', 'RESULT_FILE': 'file.txt', 'REPORT_STEPS': '3'}
```
Comment (mortalisk) [inline src/ert/_c_wrappers/enkf/ensemble_config.py:52]: This is repeated from above

File: src/ert/_c_wrappers/enkf/ensemble_config.py [hunk 2]
```diff
@@ -52,9 +76,35 @@ def _option_dict(option_list: List[str], offset: int) -> Dict[str, str]:
...
+    True
+    >>> _str_to_bool("FALSE")
+    False
+    >>> _str_to_bool("false")
+    False
+    >>> _str_to_bool("False")
+    False
+
+    Any text which is not correctly identified as true or false returns False, but
+    with failure message is logged:
```
Comment (mortalisk) [inline src/ert/_c_wrappers/enkf/ensemble_config.py:102]: Fix that sentence

File: src/ert/_c_wrappers/enkf/ensemble_config.py [hunk 3]
```diff
@@ -217,7 +267,7 @@ def gen_kw_node(gen_kw: Union[dict, list], tag_format: str) -> EnkfConfigNode:
             tmpl_path = _get_abs_path(gen_kw[1])
             out_file = _get_filename(_get_abs_path(gen_kw[2]))
             param_file_path = _get_abs_path(gen_kw[3])
-            forward_init = _str_to_bool(options.get(ConfigKeys.FORWARD_INIT))
+            forward_init = _str_to_bool(options.get(ConfigKeys.FORWARD_INIT, "FALSE"))
```
Comment (mortalisk) [inline src/ert/_c_wrappers/enkf/ensemble_config.py:270]: What is this?

File: src/ert/_c_wrappers/enkf/ensemble_config.py (hunk 2 above)
Comment (eivindjahren) [reply src/ert/_c_wrappers/enkf/ensemble_config.py]: Fixed.

File: tests/unit_tests/c_wrappers/res/config/test_config_parser.py [hunk 4]
```diff
@@ -314,49 +313,3 @@ def test_valid_string():
     assert ContentTypeEnum.CONFIG_BOOL.convert_string("True")
     assert not ContentTypeEnum.CONFIG_BOOL.convert_string("False")
     assert not ContentTypeEnum.CONFIG_BOOL.convert_string("F")
-
-
-@pytest.mark.parametrize(
-    "option_list, offset, expected",
-    [
-        (["a:1"], 0, {"a": "1"}),
```
Comment (mortalisk) [inline tests/unit_tests/c_wrappers/res/config/test_config_parser.py:322]: We dont care about these. So thats fine.

**Review by mortalisk — APPROVED** [review]

**Pull request merged**